srf_generation/input_file_generation/realisation_to_srf.py

```diff
@@ -392,6 +393,161 @@ def get_n(fault_size, sub_fault_size):
     )
 
 
+def create_multi_plane_srf(
+    realisation_file: str,
+    parameter_dictionary: Dict[str, Any],
+    stoch_file: Union[None, str] = None,
+    logger: Logger = qclogging.get_basic_logger,
+):
+    name = parameter_dictionary.get("name")
+    logger.info(f"Generating srf for realisation {name}")
+
+    # pops
+    magnitude = parameter_dictionary.pop("magnitude")
+    rake = parameter_dictionary.pop("rake")
+    dip = parameter_dictionary.pop("dip")
+    dt = parameter_dictionary.pop("dt", 0.005)
+    dtop = parameter_dictionary.pop("dtop")
+    shypo = parameter_dictionary.pop("shypo")
+    dhypo = parameter_dictionary.pop("dhypo")
+    genslip_version = str(parameter_dictionary.pop("genslip_version"))
+    dip_dir = parameter_dictionary.pop("dip_dir")
+    seed = parameter_dictionary.pop("srfgen_seed")
+    tect_type = parameter_dictionary.pop("tect_type")
+    plane_count = parameter_dictionary.pop("plane_count")
+
+    strike = [
+        parameter_dictionary.pop(f"strike_subfault_{i}") for i in range(plane_count)
+    ]
+    flen = [
+        parameter_dictionary.pop(f"length_subfault_{i}") for i in range(plane_count)
+    ]
+    fwid = [parameter_dictionary.pop(f"width_subfault_{i}") for i in range(plane_count)]
+
+    clon = [parameter_dictionary.pop(f"clon_subfault_{i}") for i in range(plane_count)]
+    clat = [parameter_dictionary.pop(f"clat_subfault_{i}") for i in range(plane_count)]
+
+    # gets
+    vel_mod_1d = parameter_dictionary.get("vel_mod_1d", DEFAULT_1D_VELOCITY_MODEL_PATH)
+    rvfac = parameter_dictionary.get("rvfac", None)
+
+    logger.debug(
+        "All srf generation parameters successfully obtained from the realisation file"
+    )
+
+    dlen = dwid = 0.1
```

Comment:
hard coded within function? I know 0.1 has been the standard value we've used for a while but is there any reason that we want to force them to always be constant?

Reply:
I don't see us increasing that resolution for the time being. So probably fine as a declared constant.